presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -579,42 +595,101 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
             return output;
         }
 
-        private DynamicFiltersResult createDynamicFilters(JoinNode node, List<JoinNode.EquiJoinClause> equiJoinClauses, Session session, PlanNodeIdAllocator idAllocator)
+        private DynamicFiltersResult createDynamicFilters(
+                JoinNode node,
+                List<JoinNode.EquiJoinClause> equiJoinClauses,
+                List<Expression> joinFilterClauses,
+                Assignments.Builder leftProjections,
+                Assignments.Builder rightProjections,
+                Session session, PlanNodeIdAllocator idAllocator)
         {
             if ((node.getType() != INNER && node.getType() != RIGHT) || !isEnableDynamicFiltering(session) || !dynamicFiltering) {
-                return new DynamicFiltersResult(ImmutableMap.of(), ImmutableList.of());
-            }
+                return new DynamicFiltersResult(ImmutableMap.of(), ImmutableList.of(), joinFilterClauses, false);
+            }
+
+            AtomicBoolean newProjectionAdded = new AtomicBoolean(false);
+            ImmutableList.Builder<ComparisonExpression> comparisonClausesBuilder = ImmutableList.builder();
+            equiJoinClauses
+                    .stream()
+                    .map(clause -> new ComparisonExpression(EQUAL, clause.getLeft().toSymbolReference(), clause.getRight().toSymbolReference()))
+                    .forEach(comparisonClausesBuilder::add);
+            joinFilterClauses = joinFilterClauses.stream()
+                    .map(clause -> {
+                        if (!isJoinComparison(clause, node.getLeft().getOutputSymbols(), node.getRight().getOutputSymbols(), DYNAMIC_FILTERING_SUPPORTED_OPERATORS)) {
+                            return clause;
+                        }
+                        ComparisonExpression comparison = (ComparisonExpression) clause;
+                        boolean alignedComparison = node.getLeft().getOutputSymbols().containsAll(SymbolsExtractor.extractUnique(comparison.getLeft()));
+                        Expression leftExpression = (alignedComparison) ? comparison.getLeft() : comparison.getRight();
+                        Expression rightExpression = (alignedComparison) ? comparison.getRight() : comparison.getLeft();
+                        ComparisonExpression.Operator operator = alignedComparison ? comparison.getOperator() : comparison.getOperator().flip();
+
+                        Symbol leftSymbol = symbolForExpression(leftExpression);
+                        if (!node.getLeft().getOutputSymbols().contains(leftSymbol)) {
```

Comment:
ok, so this actually adds new symbols... but DF won't work for `l1 + 10` anyway, so we can only support cases where left side consist of single symbol

Reply:
Sounds good, fixed the code to support only single symbol case.